dvc/api.py

```diff
@@ -45,7 +45,10 @@ class SummonError(DvcException):
 
 
 def get_url(path, repo=None, rev=None, remote=None):
-    """Returns an url of a resource specified by path in repo"""
+    """
+    Returns the full URL to the data artifact specified by its `path` in a
+    `repo`.
```

Comment:
I would suggest noting here that it does not guarantee that URL exists.
Related #3155

Reply:
I don't like backticks, we don't use them anywhere else. Python stdlib doesn't use them either.

Reply:
any suggestions? is there a standard way of doing markup that is understood by the tools that generate a view for this (like F1 in PyCharm, quick help in VS Code, docs generators, etc)

btw, we do use them as far as I can tell internally, also, Markdown markup or some markup is being used in other libraries for external APIs

Python stdlib:

```
Return whether an object is an instance of a class or of a subclass thereof.
    
    A tuple, as in ``isinstance(x, (A, B, ...))``, may be given as the target to
    check against. This is equivalent to ``isinstance(x, A) or isinstance(x, B)
    or ...`` etc.
```

double backtics to highlight the code (and probably make it linkable).

Am I missing something here?

Reply:
@shcheklein , there are plenty _docstring_ formats: https://stackoverflow.com/questions/3898572/what-is-the-standard-python-docstring-format 

From [PEP 287](https://www.python.org/dev/peps/pep-0287/):
> Text enclosed in single backquotes is recognized as "interpreted text", whose interpretation is application-dependent. In the context of a Python docstring, the default interpretation of interpreted text is as Python identifiers. The text will be marked up with a hyperlink connected to the documentation for the identifier given

Not sure if any particular IDE takes advantages of such feature.

> we don't use them anywhere else

@Suor, we actually do :sweat_smile:  but it is quite inconsistent: `grep -R '`.*`' dvc`

--- 

I'd say that there's no need to block a PR for this. I created a separate issue to discuss the formatting across the code base: https://github.com/iterative/dvc/issues/3176

Reply:
> Not sure if any particular IDE takes advantages of such feature.

yep, all of those I've used at least.

Reply:
> it does not guarantee that URL exists
OK, noted in 2b818e20 @pared.

Reply:
> we don't use them anywhere else

@Suor are there docstrings anywhere else? In any case, the API is user-facing so we probably don't need to compare it with the rest of the code in terms of comments/docstrings. And we do use back ticks in all of the `dvc` command output referring to files, commands, cmd options, etc. See #2793

Reply:
UPDATE: OK, I see Ramón extracted the current use of docstrings and backquotes in the rest of the core repo to #3176. Thanks!

Reply:
> Text enclosed in single backquotes is recognized as "interpreted text"... In the context of a Python docstring, the default interpretation of interpreted text is as Python identifiers. The text will be marked up with a hyperlink connected to the documentation for the identifier given

This is what I was going for here. So keep it for now guys? I don't initially have a strong opinion.

>  there a standard way of doing markup that is understood by the tools that generate a view for this (like F1 in PyCharm, quick help in VS Code, docs generators, etc)

Yeah great question @shcheklein! I'll look it up.

Reply:
> I'll look it up.

Ok so for doc generators, as probably all of you know, [Sphinx](http://www.sphinx-doc.org/en/master/) is by far the most popular, with [pdoc](https://pdoc3.github.io/pdoc) a far 2nd place AFAICS.

Sphinx doesn't support Markdown. actually, but rather[reStructuredText](https://docutils.sourceforge.io/docs/user/rst/quickref.html). BTW, backquotes in reST are indeed used for interpreted text, among other uses.

**pydoc** is interesting because it supports md and [other interesting formats](https://pdoc3.github.io/pdoc/doc/pdoc/#supported-docstring-formats) worth checking out actually (from Numpy and Google).

> A good general guide about docstring options is in https://docs.python-guide.org/writing/documentation/

> An interesting VSCode extension to help write [docstrings](https://www.python.org/dev/peps/pep-0257/) in all of the above formats is https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring

Still need to research how IDE cod completion works...

Reply:
> Still need to research how IDE code completion works...

VSCode [intellisense](https://code.visualstudio.com/docs/editor/intellisense#_intellisense-features) doesn't interpret any format in the docstrings it seems. It mostly works by building and inspecting the Python code by itself.

![image](https://user-images.githubusercontent.com/1477535/72658287-5dbedb00-3974-11ea-82e0-e5ed1a85f298.png)

![image](https://user-images.githubusercontent.com/1477535/72658289-6f07e780-3974-11ea-86ef-8b263a632ed1.png)

And from what I can see [here](https://www.jetbrains.com/help/pycharm/auto-completing-code.html) PyCharm works similarly.

So for IDEs, easy to read docstrings e.g. [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) or [Google style](http://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings) may be best.

Reply:
p.s. numpydoc uses single backticks to enclose variables. Google doesn't seem to use them at all though.

Reply:
Extracted docstring format discussion fully to https://github.com/iterative/dvc/issues/3176#issuecomment-575927160

Reply:
So to resolve this one @Suor, as most available formats use backquotes except Google, are you able to approve this for the time being? We will revisit this once #3176 is resolved anyway 🙂